Unicode/ustring.c

```diff
@@ -846,7 +866,7 @@ long utf82u_strlen(const char *utf8_str) {
     while ( (ch = utf8_ildb(&utf8_str))>0 && ++len>0 )
 	if ( ch>=0x10000 )
 	    ++len;
-    return( len );
+    return( len < 0 ? -1 : len );
```

Comment:
why

Reply:
was thinking about utype.c where we also have -1

Reply:
If that never happens here, we ought to remove the check since it is bound to confuse people.